test-loop-tests/src/tests/indexer.rs

```diff
@@ -192,20 +211,73 @@ fn shutdown(env: TestLoopEnv) {
     env.shutdown_and_drain_remaining_events(Duration::seconds(20));
 }
 
+fn next_nonce() -> Nonce {
+    static NEXT_VALUE: AtomicU64 = AtomicU64::new(1);
+    NEXT_VALUE.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
+}
+
+fn tx_block_hash(env: &TestLoopEnv) -> CryptoHash {
+    TestLoopNode::rpc(&env.node_datas).head(env.test_loop_data()).last_block_hash
+}
+
+fn user_signer() -> Signer {
+    create_user_test_signer(&user_account())
+}
+
+fn send_validator_tx(env: &TestLoopEnv, tx: SignedTransaction) {
+    let process_tx_request =
+        ProcessTxRequest { transaction: tx, is_forwarded: false, check_only: false };
+    TestLoopNode::validator(&env.node_datas, 0).data().rpc_handler_sender.send(process_tx_request);
+}
+
 fn create_local_tx(env: &TestLoopEnv) -> SignedTransaction {
-    let rpc_node = TestLoopNode::rpc(&env.node_datas);
-    let block_hash = rpc_node.head(env.test_loop_data()).last_block_hash;
-    let nonce = 1;
-    let tx = SignedTransaction::call(
-        nonce,
+    SignedTransaction::call(
+        next_nonce(),
         user_account(),
         user_account(),
-        &create_user_test_signer(&user_account()),
+        &user_signer(),
         Balance::ZERO,
         "does_not_matter".to_owned(),
         vec![],
         Gas::from_teragas(100),
-        block_hash,
+        tx_block_hash(env),
+    )
+}
+
+fn create_burn_gas_tx(env: &TestLoopEnv) -> SignedTransaction {
+    let gas_to_burn = GAS_LIMIT.checked_div(2).unwrap().checked_add(Gas::from_gas(1)).unwrap();
+    SignedTransaction::call(
+        next_nonce(),
+        user_account(),
+        user_account(),
+        &user_signer(),
+        Balance::ZERO,
+        "burn_gas_raw".to_owned(),
+        gas_to_burn.as_gas().to_le_bytes().to_vec(),
+        GAS_LIMIT,
+        tx_block_hash(env),
+    )
+}
+
+fn generate_delayed_receipt(env: &mut TestLoopEnv) -> BlockHeight {
+    // Each transaction generates local receipt consuming more than a half
+    // the chunk space, so exactly one receipt execution will be delayed.
+    for _ in 0..3 {
+        send_validator_tx(&env, create_burn_gas_tx(&env));
+    }
+    let validator_node = TestLoopNode::validator(&env.node_datas, 0);
+    validator_node.run_for_number_of_blocks(&mut env.test_loop, 3);
+    let delayed_receipt_executed_height = validator_node.head(env.test_loop_data()).height;
```

Comment:
nit: Is there a way to validate that our setup was successful and delayed receipts were actually produced at this height?